darity_test.go

```diff
@@ -1,13 +1,45 @@
+// +build linux
+
+// These tests use a mocked KVM virtual device and allow different ioctl
+// implementations to be used.
+
 package darity
 
 import (
+	"io/ioutil"
+	"os"
 	"testing"
 )
 
-func TestAPIVersion(t *testing.T) {
-	v, err := APIVersion()
+// TestAPIVersion verifies that Client.APIVersion returns the
+// KVM API version identified by the Version constant.
+func TestAPIVersionKVM(t *testing.T) {
+	c := &Client{
+		ioctl: func(fd uintptr, request int, argp uintptr) (int, error) {
+			return Version, nil
+		},
+	}
+
+	v, err := c.APIVersion()
 	if err != nil {
 		t.Errorf("could not get API version: %q", err.Error())
 	}
-	t.Logf("APIVersion: %d", v)
+
+	if want, got := v, Version; want != got {
```

Comment:
The mock here is very cool for getting tests to work in Travis but are we really testing more than the Go runtime with this?

I suppose this could all be extended to check that the correct ioctl numbers are passed and such. We should start working out to get real integration tests running, too.